### PR TITLE
feat(looker): index looker charts and dashboards by business term

### PR DESCRIPTION
**Pull request by gabe-lyons**
image

**Metadata Model Changes**
- Adds a new aspect InputFields for charts and dashboards. This tracks the inputs a chart uses so that relevant charts will be surfaced in search. It also stores column-level lineage information.
- Adds a property `label` to the SchemaField model. This is an optional string for giving a more readable name to fields. It is currently not being displayed in the UI.

**Ingestion Changes**
- Adds logic in looker to produce InputFields aspects for charts & dashboards, and to add the label property to views & explores.

**Product Changes**
- Adds logic to show search snippets for charts and dashboards that are discovered by their input fields.

## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-models/src/main/pegasus/com/linkedin/common/InputField.pdl [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  schemaField: optional SchemaField
+
+  /**
+   * Urn of the schema being referenced for lineage purposes
+   */
+  @Relationship = {
+    "name": "consumesField",
+    "entityTypes": [ "schemaField" ]
+  }
+  schemaFieldUrn: optional Urn
```
Comment (shirshanka) [inline metadata-models/src/main/pegasus/com/linkedin/common/InputField.pdl:21]: should not be optional?

File: metadata-models/src/main/pegasus/com/linkedin/common/InputField.pdl [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import com.linkedin.schema.SchemaField
+
+/**
+ * Information about a field a chart or dashboard references
+ */
+record InputField {
+  /**
+   * Copied version of the referenced schema field object for indexing purposes
+   */
+  schemaField: optional SchemaField
```
Comment (shirshanka) [inline metadata-models/src/main/pegasus/com/linkedin/common/InputField.pdl:12]: move this field below the urn field, just from a readability perspective, the urn is the primary thing and this is a maybe filled out thing for indexing.

File: metadata-ingestion/src/datahub/ingestion/source/looker_common.py [hunk 3]
```diff
@@ -644,6 +661,7 @@ def from_api(  # noqa: C901
                             view_fields.append(
                                 ViewField(
                                     name=measure_field.name,
+                                    label=dim_field.label_short,
```
Comment (shirshanka) [inline metadata-ingestion/src/datahub/ingestion/source/looker_common.py:664]: typo: s/dim_field/measure_field

File: metadata-ingestion/src/datahub/ingestion/source/looker.py [hunk 4]
```diff
@@ -557,38 +672,49 @@ def _get_looker_dashboard_element(  # noqa: C901
                         LookerExplore(model_name=element.look.query.model, name=exp)
                         for exp in explores
                     ],
-                    upstream_fields=fields,
+                    upstream_fields=upstream_fields,
```
Comment (shirshanka) [inline metadata-ingestion/src/datahub/ingestion/source/looker.py:675]: curious -> what is the diff between an upstream field and an input field?

Comment (jjoyce0510) [reply metadata-ingestion/src/datahub/ingestion/source/looker.py:675]: +1 we need to be very clear about this

Comment (gabe-lyons) [reply metadata-ingestion/src/datahub/ingestion/source/looker.py]: simplified things so we just track input fields now- input fields are a superset of upstream fields and also chart-specific fields

File: metadata-ingestion/src/datahub/ingestion/source/looker_common.py (hunk 3 above)
Comment (gabe-lyons) [reply metadata-ingestion/src/datahub/ingestion/source/looker_common.py]: fixed!

File: metadata-models/src/main/pegasus/com/linkedin/common/InputField.pdl (hunk 2 above)
Comment (gabe-lyons) [reply metadata-models/src/main/pegasus/com/linkedin/common/InputField.pdl]: done

File: metadata-models/src/main/pegasus/com/linkedin/common/InputField.pdl (hunk 1 above)
Comment (gabe-lyons) [reply metadata-models/src/main/pegasus/com/linkedin/common/InputField.pdl]: updated

File: metadata-ingestion/src/datahub/ingestion/source/looker.py [hunk 5]
```diff
@@ -437,76 +452,110 @@ def _get_fields_from_query(self, query: Optional[Query]) -> List[str]:
...
-                custom_field_to_underlying_field[field]
-                if field in custom_field_to_underlying_field
-                else field
-            )
-            if field_name is None:
+            if field is None:
                 continue
 
-            all_fields.add(field_name)
+            # we haven't loaded in metadata about the explore yet, so we need to wait until explores are populated later to fetch this
```
Comment (shirshanka) [inline metadata-ingestion/src/datahub/ingestion/source/looker.py:515]: need to understand this

**Review by shirshanka — APPROVED** [review]
LGTM!

**Comment by shirshanka** [comment]
Putting this PR on hold until we address serialization of optional elements.

**Review by shirshanka — APPROVED** [review]
Good to merge now!

**Pull request merged**